lib/improver/tests/nbhood/use_nbhood/test_ApplyNeighbourhoodProcessingWithAMask.py

```diff
@@ -0,0 +1,166 @@
+               "topographic_zone, neighbourhood_method: square, radii: 2000, "
+               "lead_times: None, ens_factor: 1.0, weighted_mode: True, "
+               "sum_or_fraction: fraction>")
+        self.assertEqual(result, msg)
+
+
+class Test_process(IrisTest):
+
+    """Test the process method of ApplyNeighbourhoodProcessingWithAMask."""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5)
+        self.cube = iris.util.squeeze(self.cube)
+        mask_data = np.array([[[1, 0, 0, 0, 0],
+                               [1, 1, 0, 0, 0],
+                               [1, 1, 0, 0, 0],
+                               [0, 0, 0, 0, 0],
+                               [0, 0, 0, 0, 0]],
+                              [[0, 0, 0, 0, 0],
+                               [0, 0, 1, 1, 0],
+                               [0, 0, 1, 1, 0],
+                               [0, 0, 0, 0, 0],
+                               [0, 0, 0, 0, 0]],
+                              [[0, 0, 0, 0, 0],
+                               [0, 0, 0, 0, 0],
+                               [0, 0, 0, 0, 0],
+                               [0, 0, 0, 1, 1],
+                               [0, 0, 0, 1, 1]]])
+        topographic_zone_points = [50, 150, 250]
+        topographic_zone_bounds = [[0, 100], [100, 200], [200, 300]]
+
+        mask_cubes = iris.cube.CubeList([])
+        for data, point, bounds in zip(mask_data, topographic_zone_points,
+                                       topographic_zone_bounds):
+            mask_cubes.append(
+                set_up_topographic_zone_cube(
+                    data, point, bounds, num_grid_points=5))
+        self.mask_cube = mask_cubes.merge_cube()
+
+    def test_basic(self):
+        """Test that the expected result is returned, when the
+        topographic_zone coordinate is iterated over."""
+        expected = np.array(
+            [[[[1.00, 0.00, 0.00, 0.00, 0.00],
+               [1.00, 1.00, 0.00, 0.00, 0.00],
+               [1.00, 1.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00]]],
+             [[[0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.75, 0.75, 0.00],
+               [0.00, 0.00, 0.75, 0.75, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00]]],
+             [[[0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 0.00, 0.00],
+               [0.00, 0.00, 0.00, 1.00, 1.00],
+               [0.00, 0.00, 0.00, 1.00, 1.00]]]])
+        coord_for_masking = "topographic_zone"
+        neighbourhood_method = "square"
+        radii = 2000
+        result = ApplyNeighbourhoodProcessingWithAMask(
+            coord_for_masking, neighbourhood_method, radii
+            ).process(self.cube, self.mask_cube)
```

Comment:
This unittest should fail. neighbourhood_method is not an argument for ApplyNeighbourhoodProcessingWithAMask

Reply:
Removed.